Sources/FoundationEssentials/String/String+Internals.swift

```diff
@@ -138,19 +138,20 @@ extension String {
             return try block(buffer.baseAddress!)
         }
         #else
-        var iter = self.utf8.makeIterator()
 #if os(Windows)
+        var iter = self.utf8.makeIterator()
         let bLeadingSlash = if iter.next() == ._slash, iter.next()?.isLetter ?? false, iter.next() == ._colon { true } else { false }
-#else
-        let bLeadingSlash = false
-#endif
-
         // Strip the leading `/` on a RFC8089 path (`/[drive-letter]:/...` ).  A
         // leading slash indicates a rooted path on the drive for the current
         // working directory.
-        return try Substring(self.utf8.dropFirst(bLeadingSlash ? 1 : 0)).withCString {
+        return try Substring(self.utf8.dropFirst(bLeadingSlash ? 1 : 0)).replacing(._slash, with: ._backslash).withCString {
```

Comment:
Ah sorry one last thing. I think getting the substring directly from `self` using `try self[utf8.index(startIndex, offsetBy: bLeadingSlash ? 1 : 0)...].replacing(...` is more efficient because it doesn't create and mutate a UTF8 array and create a new substring from that. (I could be wrong or it might be compiled out, though.) This would apply for `mut` below as well

Reply:
I think it should effectively be the same because `self.utf8.dropFirst(_:)` returns a `Substring.UTF8View` (so it's just a substring's view into the original storage - this is O(1)), and then I believe the `Substring(Substring.UTF8View)` initializer just "adopts" the UTF8 view as its backing store (also O(1)). So performance wise I'd expect these to be equivalent since they should all effectively be constant time (and won't trigger a copy of the UTF-8 buffer until the call to `replacing(_:with:)`

Reply:
Oh oops you're totally right that makes sense. I just tested it and the performance is the exact same :)